### PR TITLE
Easy override for social logins

### PR DESCRIPTION
**Pull request by gangefors**
The `MAILMAN_WEB_SOCIAL_AUTH` list contains a default set of social
login provides. This was previously included in `INSTALLED_APPS`.

Separating it to it's own list makes it easier to disable or otherwise
override which social login providers are enabled without the need to
modify `INSTALLED_APPS`.

Older installations where `INSTALLED_APPS` were overridden continues to
work. The `MAILMAN_WEB_SOCIAL_AUTH` is ignored for those and
`INSTALLED_APPS` is used as is.

Solves: #305

### CODE REVIEW

**Review by maxking — CHANGES_REQUESTED** [review]
There is just a small change that is probably outside of this PR but would be good if you can remove while you are at it :)

Also, the CI is failing the head for reasons I haven't had time to investigate, so I can't merge it until the CI is fixed but the changes look good to me. I'll merge them when I can get the CI running correctly.

File: web/mailman-web/settings.py
Comment (gangefors) [reply web/mailman-web/settings.py]: I've added a commit where this is removed which is included in this PR now.

**Pull request merged**
